### PR TITLE
Utilize BucketObjectEntry in GC logic

### PR DESCRIPTION
**Pull request by jonasfj**

### CODE REVIEW

File: app/lib/package/api_export/exported_api.dart [hunk 1]
```diff
@@ -452,14 +449,14 @@ const _validatedCustomHeader = 'validated';
 ///
 /// This allows us to detect files that are present, but not being updated
 /// anymore. We classify such files as _stray files_ and write alerts to logs.
-const _updateValidatedAfter = Duration(days: 1);
+const _updateValidatedAfter = Duration(days: 3);
```
Comment (jonasfj) [inline app/lib/package/api_export/exported_api.dart:452]: Idea being that we don't want to update meta-data all the time.

File: app/lib/package/api_export/exported_api.dart [hunk 2]
```diff
@@ -98,40 +98,37 @@ final class ExportedApi {
   Future<void> _gcPrefix(String prefix, Set<String> allPackageNames) async {
```
Comment (jonasfj) [inline app/lib/package/api_export/exported_api.dart:98]: Change here simplified a bit, just list all files, extract package name from the object name and if the package doesn't exist, we delete it if it's older than `_minGarbageAge`

File: app/lib/package/api_export/exported_api.dart [hunk 3]
```diff
@@ -26,7 +26,7 @@ final _log = Logger('api_export:exported_bucket');
 /// This ensures that we don't delete files we've just created.
 /// It's entirely possible that one process is writing files, while another
 /// process is running garbage collection.
-const _minGarbageAge = Duration(days: 1);
+const _minGarbageAge = Duration(hours: 3);
```
Comment (jonasfj) [inline app/lib/package/api_export/exported_api.dart:29]: This shouldn't be high, if something is published, it'll be at-least 3 hours before we can delete it.

I think that's perhaps somewhat reasonable. But we don't want this to be too high.

However, 3 hour should be more than enough for any eventual consistency issues between datastore and GCS to have been resolved.

File: app/lib/package/api_export/exported_api.dart [hunk 4]
```diff
@@ -452,14 +449,14 @@ const _validatedCustomHeader = 'validated';
...
-const _updateValidatedAfter = Duration(days: 1);
+const _updateValidatedAfter = Duration(days: 3);
 
 /// Duration after which a file that haven't been updated is considered stray!
 ///
 /// We don't delete stray files, because there shouldn't be any, so a stray file
 /// is always indicative of a bug. Nevertheless, we write alerts to logs, so
 /// that these inconsistencies can be detected.
-const _unvalidatedStrayFileAfter = Duration(days: 7);
+const _unvalidatedStrayFileAfter = Duration(days: 12);
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/package/api_export/exported_api.dart:459]: I'm wondering: will we look at the logs or have alerts that fire within this period?

**Review by isoos — APPROVED** [review]

File: app/lib/package/api_export/exported_api.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/package/api_export/exported_api.dart:459]: No, after 12 days we'll consider a file stray, and then we'll see it in logs.

**Pull request merged**